homeassistant/components/http/auth.py

```diff
@@ -57,18 +57,23 @@ def async_sign_path(
         else:
             refresh_token_id = hass.data[STORAGE_KEY]
 
+    url = URL(path)
     now = dt_util.utcnow()
     encoded = jwt.encode(
         {
             "iss": refresh_token_id,
-            "path": unquote(path),
+            "path": url.path,
```

Comment:
Let's also include the query params in the signature and verify that they are the same.

Reply:
That is a change in functionality and the other side that validates will have to be updated as well.